core/libp2p/config.hpp

```diff
@@ -50,7 +50,8 @@ namespace libp2p {
     detail::vecsptr<security::SecurityAdaptor> securities;
     detail::vec<multi::Multiaddress> listen_addresses;
 
-    detail::sptr<boost::asio::io_context> io_context;
+    detail::sptr<boost::asio::io_context> context;
+    detail::sptr<boost::asio::io_context::executor_type> executor;
```

Comment:
Why do we need both, if each of them can be got from the other?

Reply:
All components get executor by reference, so it has to be stored somewhere, and the best place is here.